src/types.js

```diff
@@ -53,6 +53,22 @@ export function toArray(arrayLike) {
 }
 
 /**
+ * Returns a map-like object.
+ * If opt_initial is provided, copies its own properties into the
+ * newly created object.
+ * @param {T=} opt_initial This should typically be an object literal.
+ * @return {T}
```

Comment:
Yeah. This is as good as we can do in closure compiler. The results is:

- If you pass a literal, then the return value is the inferred record type of the literal.
- If you pass nothing, then the return value is `?` (sadly)